arch/x86/src/boundary/boundary_impl.rs

```diff
@@ -203,17 +165,10 @@ impl UserspaceKernelBoundary for Boundary {
             SYSCALL_VECTOR => {
                 let num = state.eax as u8;
 
-                // Syscall arguments are passed on the stack using cdecl convention.
-                //
-                // Safety: Caller of this function has guaranteed that the memory region is valid.
-                let arg0 =
-                    unsafe { state.read_stack(0, accessible_memory_start, app_brk) }.unwrap_or(0);
-                let arg1 =
-                    unsafe { state.read_stack(1, accessible_memory_start, app_brk) }.unwrap_or(0);
-                let arg2 =
-                    unsafe { state.read_stack(2, accessible_memory_start, app_brk) }.unwrap_or(0);
-                let arg3 =
-                    unsafe { state.read_stack(3, accessible_memory_start, app_brk) }.unwrap_or(0);
+                let arg0 = state.ebx;
+                let arg1 = state.ecx;
+                let arg2 = state.edx;
+                let arg3 = state.edi;
```

Comment:
Any basis on _why_ `ebx`, `ecx`, `edx`, `edi` are used in that order? Were they chosen at random or is there some comparison to other systems?

Reply:
Can you point to documentation for the Linux x86 32-bit ABI somewhere? I was looking for it but can't find anything.

The common convention I found was that `%eax`, `%edx`, and `%ecx` are caller-saved. Only three would be insufficient for our purposes though. And I guess you can just list whatever (barring LLVM rules) as clobbered registers.

Reply:
That's actually a really nice list of argument and return registers. https://man7.org/linux/man-pages/man2/syscall.2.html